firewood/tests/merkle.rs

```diff
@@ -5,10 +5,11 @@ use firewood::{
     merkle::Node,
     merkle_util::{new_merkle, DataStoreError, MerkleSetup},
     proof::ProofError,
+    // TODO: we should not be using shale from an integration test
```

Comment:
Do you mean, "for an integration test"? I think an integration test *is* the place to use shale, isn't it? It's in the unit tests that we don't want to be using these types (but we're always tied to a `ShaleStore` implementation regardless.

Reply:
No. Integration tests test the public interface. We might be able to use it from a functional test though.